examples/sparse/bal_loader.py

```diff
@@ -0,0 +1,254 @@
+            The camera extrinsics.
+            First three columns are translation, last four columns is unit quaternion.
+        - camera_intrinsics: torch.Tensor (n_cameras, 3, 3)
+            The camera intrinsics. Each camera is represented as a 3x3 K matrix.
+        - camera_distortions: torch.Tensor (n_cameras, 2)
+            The camera distortions. k1 and k2.
+        - points_3d: torch.Tensor (n_points, 3)
+            contains initial estimates of point coordinates in the world frame.
+        - points_2d: torch.Tensor (n_observations, 2)
+            contains measured 2-D coordinates of points projected on images in each observations.
+        - camera_index_of_observations: torch.Tensor (n_observations,)
+            contains indices of cameras (from 0 to n_cameras - 1) involved in each observation.
+        - point_index_of_observations: torch.Tensor (n_observations,)
+            contains indices of points (from 0 to n_points - 1) involved in each observation.
+    """
+    with open(file_name, "r") as file:
+        n_cameras, n_points, n_observations = map(
+            int, file.readline().split())
+
+        camera_indices = np.empty(n_observations, dtype=int)
+        point_indices = np.empty(n_observations, dtype=int)
+        points_2d = np.empty((n_observations, 2), dtype=np.float32)
+
+        for i in range(n_observations):
+            camera_index, point_index, x, y = file.readline().split()
+            camera_indices[i] = int(camera_index)
+            point_indices[i] = int(point_index)
+            points_2d[i] = [float(x), float(y)]
+        points_2d = torch.from_numpy(points_2d)
+
+        camera_params = np.empty(n_cameras * 9, dtype=np.float32)
+        for i in range(n_cameras * 9):
+            camera_params[i] = float(file.readline())
+        camera_params = camera_params.reshape((n_cameras, -1))
+
+        points_3d = np.empty(n_points * 3, dtype=np.float32)
+        for i in range(n_points * 3):
+            points_3d[i] = float(file.readline())
+        points_3d = points_3d.reshape((n_points, -1))
+        points_3d = torch.from_numpy(points_3d)
+
+    # convert Rodrigues vector to unit quaternion for camera rotation
+    # camera_params[0:3] is the Rodrigues vector
+    r = Rotation.from_rotvec(camera_params[:, :3])
+    q = r.as_quat()
+
+    # use pp.LieTensor of shape (n_cameras, 7) to represent camera extrinsics
+    # camera_params[3:6] is the camera translation
+    camera_extrinsics = pp.LieTensor(np.concatenate([camera_params[:, 3:6], q], axis=1), ltype=pp.SE3_type)
+
+    # use torch.Tensor of shape (n_cameras, 3, 3) to represent camera intrinsics
+    # camera_params[6] is focal length, camera_params[7] and camera_params[8] are two radial distortion parameters
+    # the origin of the image coordinate system is at the center of the image
+    camera_intrinsics = np.zeros((n_cameras, 3, 3), dtype=np.float32)
+    camera_intrinsics[:, 0, 0] = camera_params[:, 6]
```

Comment:
Worth a comment saying that putting the distortion parameters in the intrinsic matrix is something like a hack and the customized projection error function will handle it correctly later.